app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -61,6 +63,19 @@ def update_os_from_facts
         scoped_search :relation => :host_traces, :on => :application, :complete_value => true, :rename => :trace_app, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :app_type, :complete_value => true, :rename => :trace_app_type, :only_explicit => true
         scoped_search :relation => :host_traces, :on => :helper, :complete_value => true, :rename => :trace_helper, :only_explicit => true
+
+        scoped_search relation: :pools, on: :pools_expiring_in_days, ext_method: :find_with_expiring_pools, only_explicit: true
+
+        def self.find_with_expiring_pools(_key, _operator, days_from_now)
+          host_ids = with_pools_expiring_in_days(days_from_now).ids
+          scoped_search_params = {
```

Comment:
To recreate the whole params hash based on the condition is an unusual way to write this code. Why not conditionally set the conditions key instead? I had my prior suggestion but here's another way:

```
scoped_search_params = {}
if host_ids.any? # any? is more idiomatic than present?
  scoped_search_params[:condition] = ....
else
 scoped_search_params[:condition] ...
end
```

Reply:
@jturel sorry, forgot to update that